lib/rex/ui/text/input/readline.rb

```diff
@@ -17,17 +17,18 @@ class Input::Readline < Rex::Ui::Text::Input
     #
     # Initializes the readline-aware Input instance for text.
     #
-    def initialize(tab_complete_proc = nil)
+    def initialize(tab_complete_proc)
       if(not Object.const_defined?('Readline'))
         require 'readline'
       end
 
       self.extend(::Readline)
 
-      if (tab_complete_proc)
+      if tab_complete_proc
         ::Readline.basic_word_break_characters = ""
-        ::Readline.completion_proc = tab_complete_proc
-        @rl_saved_proc = tab_complete_proc
+        ::Readline.completion_proc = wrap_tab_compete_proc(tab_complete_proc)
+
+        @rl_saved_proc = wrap_tab_compete_proc(tab_complete_proc)
```

Comment:
Looks like this wraps the _previous_ readline completion proc in a try/catch, not sure if that's what we want to happen here? 👀

Reply:
No it's wrapping the `tab_complete_proc` that gets passed in with `initialize`. The previous `tab_complete_proc` is getting overwritten here